datalad_service/migrate/versions.py

```diff
@@ -0,0 +1,209 @@
+                        # # get a json output of the properties of the filekey
+                        examinekey = ds.repo._run_annex_command(
+                            'examinekey', annex_options=[file_key, '--json'])
+                        try:
+                            # # get the json output string from examinekey
+                            keyDetails = examinekey[0].strip()
+                            # print('keyDetails:', keyDetails)
+
+                            # # get the json object associated with examinekey, so we can access hashdirlower property
+                            try:
+                                keyJson = json.loads(keyDetails)
+                                # print('keyJson:', keyJson)
+
+                                # # get the metalog hash directory path in the git-annex branch
+                                hashDirLower = keyJson['hashdirlower']
+                                # print('hashDirLower:', hashDirLower)
+
+                                updateStr = '{}s {}:V +{}#{}/{}\n'.format(
+                                    ts, annex_uuid, version, datasetId, name)
+                                # print('updateStr:', updateStr)
+
+                                path = '{}{}.log.rmet'.format(
+                                    hashDirLower, file_key)
+                                # print('path:', path)
+
+                                updateObj = {'path': path,
+                                             'contents': updateStr}
+                                # print('updateObject:', updateObj)
+
+                                updates.append(updateObj)
+                            except:
+                                print('Error: {} | Response from examinekey is not properly formatted json.'.format(
+                                    fileInfo))
+                        except:
+                            print(
+                                'Error: {} | The key details cannot be determined - examinekey has no length.'.format(fileInfo))
+                    except:
+                        print(
+                            'Error: {} | Unable to execute examinekey on this file!'.format(fileInfo))
+                else:
+                    print('Error: {} | Unable to find the annex uuid.'.format(fileInfo))
+            except:
+                print('Error: {} | Unable to find the filekey associated with this file. It is quite possible that this file is not annexed.'.format(fileInfo))
+
+        print('Success: {}:{} | Successfully generated a list of update objects for this dataset snapshot.'.format(
+            datasetId, tag))
+        # checkout the git-annex branch of datalad repo
+        try:
+            print('Checking out git-annex branch to apply updates...')
+            ds.repo._git_custom_command([], cmd_str='git checkout git-annex')
+            time.sleep(3)
+            print('Current branch: ')
+            ds.run('git branch')
```

Comment:
ds.run should not be used anywhere. This will make an extra commit.

Reply:
@nellh but it makes for such a pretty log entry!